fili-core/src/main/java/com/yahoo/bard/webservice/web/filters/BardLoggingFilter.java

```diff
@@ -199,6 +205,30 @@ public void filter(ClientRequestContext request, ClientResponseContext response)
     }
 
     /**
+     * Add a request id to pass to druid. Only added if x-request-id follows the format specified at:
+     * https://devcenter.heroku.com/articles/http-request-id
+     *
+     * @param requestId  Request id to add as queryId prefix to druid
+     */
+    private void appendRequestId(String requestId) {
+        if (isValidRequestId(requestId)) {
+            return; // Ignore according to https://devcenter.heroku.com/articles/http-request-id
+        }
+        RequestLog.addIdPrefix(requestId);
+    }
+
+    /**
+     * Validate whether or not a string is acceptable as an x-request-id header argument.
+     *
+     * @param requestId  Request id to validate
+     * @return True if valid, false otherwise
+     */
+    private boolean isValidRequestId(String requestId) {
+        return requestId == null || requestId.isEmpty() || requestId.length() > 200
+               || !VALID_REQUEST_ID.matcher(requestId).matches();
```

Comment:
It looks like this method is actually returning the opposite of what it claims. It's returning `true` if the requestId is _not_ valid, and `false` if it is valid.